compiler/ballerina-lang/src/main/java/org/wso2/ballerinalang/compiler/semantics/analyzer/Types.java

```diff
@@ -4340,16 +4343,66 @@ boolean validNumericTypeExists(BType type) {
         }
     }
 
-    private boolean checkValidNumericTypesInUnion(BType memType, int firstTypeTag) {
-        if (memType.tag != firstTypeTag && !checkTypesBelongToInt(memType.tag, firstTypeTag)) {
-            return false;
+    boolean validStringOrXmlTypeExists(BType type) {
```

Comment:
Isn't this a duplicate of `validNumericTypeExists` with just minor changes? Can't we rewrite this in a better way, minimizing duplicate code? Maybe using functional interfaces.